doc/rest_api/snippet_sharing/api-description.apib

```diff
@@ -0,0 +1,587 @@
+            - `AUTH_MISSING_USERNAME` if no input for `username` was submitted
+            - `AUTH_MISSING_PASSWORD` if no input for `password` was submitted
+
++ Response 401 (application/json)
+    + Attributes (object)
+        + status (string) - Textual description of the HTTP status
+        + message (string) - Detailed error information, e.g. hint about malformed request
+        + i18n (string) - A unique token representing above error description message; can be used for internationalization in frontends
+
+            A comprehensive list of possible errors:
+            - `AUTH_UNKNOWN_USERNAME` if there is no user registered witht he given name
+            - `AUTH_INVALID_PASSWORD` if the given password does not match the password saved for the given account
+
+
+# Users endpoint [/users{?rows}{?sort}{?sortby}{?filter}{?filterby}{?current}]
+
+## User list [GET]
+
+Retrieves a list of users. The result can be adjusted by additional, but optional parameters. This operation requires moderator permissions (rank 2) or higher.
+
++ Parameters
+    + rows (number, optional) - How many entries will be added to the output at max.
+        + Default: `200`
+    + sort (enum[string], optional) - How to sort the entries for the output.
+        + Default: `asc`
+        + Members
+            + `asc`
+            + `desc`
+    + sortby (enum[string], optional) - What criteria should be used for sorting
+        + Default: `username`
+        + Members
+            + `username` - The users name
+            + `email` - email
+            + `created_at` - Creation date
+    + filter (string, optional) - Only users containing the search string in either their name or their email will be added to the output.
+    + filterby (enum[string], optional) - On what fields the filter should be applied to (i.e. what fields to involve in search)
+        + Default: `all`
+        + Members
+            + `all` - All fields: username and email
+            + `username` - Only the username
+            + `email` - Only the email
+
++ Response 200 (application/json)
+    + Attributes (User List)
+
+## Retrieve user details [GET /{name}{?current}]
+
+Retrieves the details of the specified user. If the parameter `current` is specified, the currently authenticated user will be chosen as target. Retrieving details of other users requires moderator permissions (rank 2) or higher.
+
++ Parameters
+    + current (boolean, optional) - If the currently authenticated users details should be fetched
+        + Default: `false`
+    + name (string, required) - The `username` of the user whose details should be fetched
+        
++ Response 200 (application/json)
+    + Attributes (object)
+        + username (string) - The name of the user
+        + email (string) - The email of the user
+        + rank (enum[number]) - The rank of the user (higher means more permissions)
+            + Members
+                + `1` - Normal user
+                + `2` - Moderator
```

Comment:
Interesting idea, but for your example I'd simply create a new account and would be fine again (only solution would be to change the default group to something with less privilegues too). But I'll change the groups to something else, then we don't have to discuss it to the end now:
- 10 - normal user
- 50 - moderator
- 100 - admin